### PR TITLE
Enable App Library more natively via _os_feature_enabled_impl

### PR DESCRIPTION
**Pull request by PoomSmart**
This includes generic improvements to the codebase.

### CODE REVIEW

File: AppLibraryEnabler.xm [hunk 1]
```diff
@@ -160,29 +160,12 @@
...
-	bool origValue = %orig;
-	if ( [[self _viewControllerForAncestor] isKindOfClass:%c( SBHIconLibraryTableViewController )] || [[self _viewControllerForAncestor] isKindOfClass:%c( SBHLibraryCategoryIconViewController )] || [[self _viewControllerForAncestor] isKindOfClass:%c( SBHLibraryPodCategoryFolderController )] ) {
-		NSMutableDictionary *defaults = [NSMutableDictionary dictionaryWithContentsOfFile:[NSString stringWithFormat:@"%@/Library/Preferences/com.apple.springboard.plist", NSHomeDirectory()]];
-		bool sbHomeScreenShowsBadgesInAppLibrary = [[defaults objectForKey:@"SBHomeScreenShowsBadgesInAppLibrary"] boolValue];
-		return sbHomeScreenShowsBadgesInAppLibrary;
-	}
-	return origValue;
+extern "C" bool _os_feature_enabled_impl(const char *domain, const char *feature);
+%hookf(bool, _os_feature_enabled_impl, const char *domain, const char *feature) {
+    if (strcmp(domain, "SpringBoard") == 0 && strcmp(feature, "Dewey") == 0)
```
Comment (PoomSmart) [inline AppLibraryEnabler.xm:165]: Protip: check out `/System/Library/FeatureFlags/Global.plist`

File: AppLibraryEnabler.xm [hunk 2]
```diff
@@ -160,29 +160,12 @@
...
+        return true;
+    return %orig;
 }
-%end
-
-%hook SBHIconManager
-- (bool)iconLocationAllowsBadging:(id)arg1 {
```
Comment (PoomSmart) [inline AppLibraryEnabler.xm:176]: I confirmed that removing these hooks doesn't affect the functionalities.

**Comment by tomaszpoliszuk** [comment]
All of those seems great - I'll check them as soon as I can on simulator and most likely accept this PR, thank you!

**Comment by tomaszpoliszuk** [comment]
I have alternative code that makes App Library full screen but it didn't worked correctly or was interfering with today view - I'll prepare it and add in separate branch if you would be interested to look into it.

Other option i was considering was to move App Library to right side of screen and make it work like today view (again - was unable to do it so far without breaking today view).

**Comment by PoomSmart** [comment]
Yeah, I came across that exact issue when I tried to modify the container of App Library. Maybe it's time to RE iOS 15 binaries and see how they do it nicely.

**Comment by tomaszpoliszuk** [comment]
I didn't tried any RE (yet) but tried to checkra1n my 6s (didn't worked) - so no headers for now.

Excerpt from that fullscreen version that (i suspect) could be helpful (don't mind `viewWillLayoutSubviews` - it should be changed to `viewDidAppear` or something else)

```
%hook SBHomeScreenOverlayViewController
- (void)viewWillLayoutSubviews {
	%orig;
	if ( [[self contentViewController].avocadoViewController isKindOfClass:%c(SBHLibraryViewController)] ) {
		[[self contentWidthConstraint] setConstant:[UIScreen mainScreen].bounds.size.width];
	} else {
		[[self contentWidthConstraint] setConstant:393];
	}
}
%end
```

I'll post that branch tomorrow.

File: AppLibraryEnabler.xm (hunk 1 above)
Comment (tomaszpoliszuk) [reply AppLibraryEnabler.xm:165]: This file does not exist in .ipsw (or it's iOS13+)? - just checked in 12.5.3.ipsw and it wasn't there

If it exist then can it be used to enable pip on iOS12?

Comment (tomaszpoliszuk) [reply AppLibraryEnabler.xm:165]: Just checked in 14.6.ipsw - it exists there :)

**Comment by tomaszpoliszuk** [comment]
Ok - it's on [fullwidth branch](https://github.com/tomaszpoliszuk/AppLibraryEnabler/tree/fullwidth)
Things on todo list at the moment:
- edit _SBHLibraryPodIconListView frame
- edit SBHIconGridSize or SBHIconGridSizeClassSizes (or something else related to IconGridSize?) in app library

**Pull request merged**